### PR TITLE
Refactor : API calls moved out of the LoginActivity

### PR DESCRIPTION
**Pull request by neeldoshii**
## Fixes

* Related #5533 

## Approach

I have moved the API logic (business logic) from login activity to login client. 

## Testing Instructions
1. Open the app (make sure you are logged out or perform log out if logged in).
2. Perform login

## Tests performed
Test performed on Google Pixel 6 API 24 (Emulator).

Feel free to place your opinion and reviews.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt [hunk 1]
```diff
@@ -106,6 +107,35 @@ class LoginClient(private val loginInterface: LoginInterface) {
...
     }
 
+    fun doLogin(
+        username: String,
+        password: String,
+        twoFactorCode: String? = null,
+        userLanguage: String = Locale.getDefault().language,
+        loginCallback: LoginCallback
+    ) {
+        getLoginToken().enqueue(object :Callback<MwQueryResponse?>{
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt:117]: If no-one outside of the `LoginClient` is calling the `getLoginToken()` method, can you make it _private_?

Comment (neeldoshii) [reply app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt:117]: Sure!! I made the changes in the commit.

File: app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt [hunk 2]
```diff
@@ -106,6 +107,35 @@ class LoginClient(private val loginInterface: LoginInterface) {
         })
     }
 
+    fun doLogin(
+        username: String,
+        password: String,
+        twoFactorCode: String? = null,
+        userLanguage: String = Locale.getDefault().language,
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt:114]: Are these default parameter values actually needed, and can we drop the import of `Locale`?  Looks like the `LoginActivity` is passing them.  Also, _nullability_ - the twoFactorCode is required to be non-null in the Activity.  You might be able to drop the "?" from it here.

File: app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt [hunk 3]
```diff
@@ -106,6 +107,35 @@ class LoginClient(private val loginInterface: LoginInterface) {
...
+            ) {
+                if (response.isSuccessful){
+                    val loginToken = response.body()?.query()?.loginToken()
+                    loginToken?.let {
+                        login(username, password, null, twoFactorCode, it, userLanguage, loginCallback)
+                    } ?: run {
+                        loginCallback.error(IOException("Failed to retrieve login token"))
+                    }
+                } else {
+                        loginCallback.error(IOException("Failed to retrieve login token"))
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt:130]: Not a big deal, but the indenting seems off 😄

**Review by psh — CHANGES_REQUESTED** [review]
Looks good to me!  Thanks for the submission.  Just a couple of comments to look at.

File: app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt (hunk 3 above)
Comment (neeldoshii) [reply app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt]: fixed the indentation. I hope adding klint will fix such issues before commit.

File: app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt (hunk 2 above)
Comment (neeldoshii) [reply app/src/main/java/fr/free/nrw/commons/auth/login/LoginClient.kt]: Fixed.

**Comment by neeldoshii** [comment]
Do i need to squash merge all my commits into one?

**Review by psh — APPROVED** [review]

**Comment by psh** [comment]
Thank you for the fixes! 😄

**Pull request merged**